src/component/common/Table/TableSearch/TableSearchField/TableSearchFieldSuggestions/TableSearchFieldSuggestions.tsx

```diff
@@ -0,0 +1,202 @@
+    color: theme.palette.text.primary,
+}));
+
+const StyledDivider = styled(Divider)(({ theme }) => ({
+    border: `1px dashed ${theme.palette.dividerAlternative}`,
+    margin: theme.spacing(1.5, 0),
+}));
+
+const StyledCode = styled('span')(({ theme }) => ({
+    backgroundColor: theme.palette.secondaryContainer,
+    color: theme.palette.text.primary,
+    padding: theme.spacing(0, 0.5),
+    borderRadius: theme.spacing(0.5),
+}));
+
+interface TableSearchFieldSuggestionsProps {
+    getSearchContext: () => IGetSearchContextOutput;
+}
+
+export const TableSearchFieldSuggestions: VFC<
+    TableSearchFieldSuggestionsProps
+> = ({ getSearchContext }) => {
+    const searchContext = getSearchContext();
+    const getSearchText = getSearchTextGenerator(searchContext.columns);
+
+    const randomRow = useMemo(
+        () => randomIndex(searchContext.data),
+        [searchContext.data]
+    );
+
+    const filters = getFilterableColumns(searchContext.columns)
+        .map(column => {
+            const filterOptions = searchContext.data.map(row =>
+                getColumnValues(column, row)
+            );
+
+            return {
+                name: column.filterName,
+                header: column.Header ?? column.filterName,
+                options: [...new Set(filterOptions)].sort((a, b) =>
+                    a.localeCompare(b)
+                ),
+                suggestedOption:
+                    filterOptions[randomRow] ?? `example-${column.filterName}`,
+                values: getFilterValues(
+                    column.filterName,
+                    searchContext.searchValue
+                ),
+            };
+        })
+        .sort((a, b) => a.name.localeCompare(b.name));
+
+    const searchableColumns = searchContext.columns.filter(
+        column => column.searchable && column.accessor
+    );
+
+    const searchableColumnsString = searchableColumns
+        .map(column => column.Header ?? column.accessor)
+        .join(', ');
+
+    const suggestedTextSearch =
+        searchContext.data.length && searchableColumns.length
+            ? getColumnValues(
+                  searchableColumns[0],
+                  searchContext.data[randomRow]
+              )
+            : 'example-search-text';
+
+    const searchText = getSearchText(searchContext.searchValue);
+    const searchFilters = filters.filter(filter => filter.values.length > 0);
+
+    return (
+        <StyledPaper>
+            <StyledBox>
+                <StyledFilterList />
+                <Box>
+                    <ConditionallyRender
+                        condition={Boolean(searchContext.searchValue)}
+                        show={
+                            <>
+                                <StyledHeader>Searching for:</StyledHeader>
+                                <ConditionallyRender
+                                    condition={Boolean(searchText)}
+                                    show={
```

Comment:
This render function is getting quite large. I think we can simplify this one by breaking some of it into subcomponents.

Reply:
Addressed in: https://github.com/Unleash/unleash-frontend/pull/1052/commits/3cc1535b3bbf991375c890a4397553481f1072c6